File: 11-ContainerWithMostWater/11-ContainerWithMostWater.java
// Last updated: 8/11/2026, 12:27:30 PM
class Solution {
    public int maxArea(int[] arr) {
        int left = 0;
        int right = arr.length - 1;
        int maxArea = 0;

        while (left < right) {
            int minHeight = Math.min(arr[left], arr[right]);
            int length = right - left;
            int area = length * minHeight;

          
            if (area > maxArea) {
                maxArea = area;
            }

            if (arr[left] < arr[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxArea;
    }
}

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 9/2/2026, 12:46:01 PM
class Solution:
    def maxArea(self, height):
        left = 0
        right = len(height) - 1
        answer = 0

        while left < right:
            width = right - left
            area = width * min(height[left], height[right])

            answer = max(answer, area)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return answer